src/PostgREST/Config.hs

```diff
@@ -108,7 +108,7 @@ dumpLogLevel = \case
   LogInfo  -> "info"
 
 data OpenAPIMode = OAFollowPriv | OAIgnorePriv | OADisabled
-  deriving Eq
```

Comment:
`Show` is not required. I found it helpful for troubleshooting and harmless to leave in. Happy to take it back out if it's a concern.

Can't at the moment remember the `Eq` bit except that some structures (and substructures) need `Eq` and `Ord` to be put into sets.

Reply:
Right, makes sense about coverage. I wish Haskell would have like a global "derive Show everywhere" flag. While I was working on this, being able to trace log stuff about the state of the db cache was helpful as someone knowing very little about the codebase.

Reply:
Ah ok, yes it would affect coverage. 

I wish Haskell had a "derive Show automatically" flag because it's a hassle to go and add it everywhere when you need to trace/debug print some stuff or you're just doing initial exploration of a codebase you haven't worked on before. Then you could just turn that on temporarily like you turn on other debug stuff while working.

Reply:
re `Show`, I've had the same issue before and would prefer all types to have it. @wolfgangwalther If we add `doctests` that use `Show`, will they be picked up by codecov?

Reply:
Call me old fashioned but I like to just debug print the state at various points when exploring or troubleshooting. So stuff like this helper method,

```
import Debug.Trace

tracePass :: ReadRequest -> Either ApiRequestError ReadRequest
tracePass v = Debug.Trace.traceShow v (Right v)
```

Inserted into a chain of processing like in `readPlan` in `Plan.hs`:

```
  tracePass =<<
  addRanges apiRequest =<<
  tracePass =<<
  addOrders apiRequest =<<
```

Now I can see how `addRanges` affects the `ReadRequest` for various inputs or unit tests.

It wouldn't work to stick it in doctests like above because I'm using it outside of doctests.

I guess one solution would be to create a source code file which does nothing but `deriving instance Show` for every type and then conditionally not compile it when doing coverage testing.

Reply:
Just dropped it all in the end. Let it be a question for some other PR.